CMakeLists.txt

```diff
@@ -27,29 +28,119 @@ set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules)
 set(CMAKE_MSVC_RUNTIME_LIBRARY MultiThreadedDLL)
 set(CMAKE_POSITION_INDEPENDENT_CODE ${BUILD_SHARED_LIBS})
 
-if(FIND_PM_DEPS)
-  find_package(SwiftSystem CONFIG REQUIRED)
-  find_package(TSC CONFIG REQUIRED)
+# Toolchain Vended dependencies
+find_package(dispatch QUIET)
+find_package(Foundation QUIET)
+
+include(FetchContent)
+
+set(_SPM_SAVED_BUILD_TESTING ${BUILD_TESTING})
+set(_SPM_SAVED_BUILD_EXAMPLES ${BUILD_EXAMPLES})
+
+set(BUILD_EXAMPLES NO)
+set(BUILD_TESTING NO)
+
+find_package(ArgumentParser CONFIG)
+if(NOT ArgumentParser_FOUND)
+  message("-- Vending swift-argument-parser")
+  FetchContent_Declare(ArgumentParser
+    GIT_REPOSITORY https://github.com/apple/swift-argument-parser
+    GIT_TAG 1.2.3)
+  FetchContent_MakeAvailable(ArgumentParser)
+endif()
+
+find_package(TSC CONFIG)
+if(NOT TSC_FOUND)
+  message("-- Vending swift-tools-support-core")
+  FetchContent_Declare(ToolsSupportCore
+    GIT_REPOSITORY https://github.com/apple/swift-tools-support-core
+    GIT_TAG main)
+  FetchContent_MakeAvailable(ToolsSupportCore)
```

Comment:
The `FetchContent_MakeAvailable` pretty explicitly calls out not calling it multiple times to ensure that transitive dependencies get resolve appropriately/consistently. It should be called once with all declared dependencies passed in at the same time.

```cmake
FetchContent_MakeAvailable(ArgumentParser ToolsSupportCore ...)
```

https://cmake.org/cmake/help/latest/module/FetchContent.html#command:fetchcontent_makeavailable

Reply:
Oh, I missed that.  Thanks!

Reply:
Build up a list of them and pass them in all at once?
```
list(APPEND VendoredDeps ArgumentParser)
...
list(APPEND VendoredDeps ToolsSupportCore)
...
FetchContent_MakeAvailable(${VendoredDeps})
```